src/player.js

```diff
@@ -1659,6 +1696,12 @@ export default class Player extends FakeEventTarget {
           }
         });
       }
+      if (this.config.playback.playAdsWithMSE) {
+        this._eventManager.listen(this, AdEventType.AD_LOADED, this._detachMediaSource);
+        this._eventManager.listen(this, AdEventType.AD_BREAK_END, this._attachMediaSource);
+        this._eventManager.listen(this, AdEventType.AD_ERROR, this._attachMediaSource);
+        this._eventManager.listen(this, AdEventType.ADS_COMPLETED, this._clearDetachTime);
```

Comment:
what is the use case for this?
I believe you can just clear it internally in the engine on ended.